tray/tray.go

```diff
@@ -196,22 +184,100 @@ func (ti *Instance) configureDebugMode() {
 	ti.debugMode = os.Getenv("NORDVPN_TRAY_DEBUG") == "1"
 }
 
+func (ti *Instance) onDaemonStateEvent(item *pb.AppState) {
+	switch st := item.GetState().(type) {
+	case *pb.AppState_Error:
+		log.Printf("%s %s Received daemon error state\n", logTag, internal.ErrorPrefix)
+		ti.updateDaemonConnectionStatus(internal.ErrDaemonConnectionRefused.Error())
+
+	case *pb.AppState_ConnectionStatus:
+		ti.updateVpnStatus()
+		ti.updateRecentConnections()
+
+	case *pb.AppState_LoginEvent:
+		ti.updateLoginStatus()
+
+	case *pb.AppState_SettingsChange:
+		ti.setSettings(st.SettingsChange)
+		// identify whether we need to also update a country list
+		ti.connSensor.Set(connectionSettings{
+			Obfuscated:      st.SettingsChange.Obfuscate,
+			Protocol:        st.SettingsChange.Protocol,
+			Technology:      st.SettingsChange.Technology,
+			VirtualLocation: st.SettingsChange.VirtualLocation,
+		})
+
+		if ti.connSensor.Detected() {
```

Comment:
Btw. isn't this always redrawing now even if the changes are not detected? I don't see the reset of `changed` field